### PR TITLE
Add GCP exporter

### PR DESCRIPTION
**Pull request by amine3**
Add GCP exporter to OpenTelemetry Mixin using opentelemetry-exporter-gcp-trace

### CODE REVIEW

File: src/ostorlab/agent/mixins/agent_open_telemetry_mixin.py [hunk 1]
```diff
@@ -12,6 +13,7 @@
 
 from opentelemetry import trace
 from opentelemetry.exporter.jaeger import thrift as jaeger
+from opentelemetry.exporter.cloud_trace import CloudTraceSpanExporter
```
Comment (mohsinenar) [inline src/ostorlab/agent/mixins/agent_open_telemetry_mixin.py:16]: don't import classes directly. 
```suggestion
from opentelemetry.exporter import cloud_trace
```

File: src/ostorlab/agent/mixins/agent_open_telemetry_mixin.py [hunk 2]
```diff
@@ -70,6 +78,13 @@ def _get_jaeger_exporter(self, parsed_url):
         logger.info('Configuring jaeger exporter..')
         return jaeger_exporter
 
+    def _get_gcp_exporter(self, parsed_url):
+        parsed_path = parsed_url.path.split(':')
+        project_id = parsed_path[0]
+        service_account_key = parsed_path[1]
+        os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = service_account_key
+        return CloudTraceSpanExporter(project_id=project_id)
```
Comment (mohsinenar) [inline src/ostorlab/agent/mixins/agent_open_telemetry_mixin.py:86]:
```suggestion
        return cloud_trace.CloudTraceSpanExporter(project_id=project_id)
```

File: src/ostorlab/agent/mixins/agent_open_telemetry_mixin.py [hunk 3]
```diff
@@ -4,6 +4,7 @@
 metadata, metrics and exceptions.
 """
 import io
+import os
```
Comment (3asm) [inline src/ostorlab/agent/mixins/agent_open_telemetry_mixin.py:7]: Where is the unit test?

File: src/ostorlab/agent/mixins/agent_open_telemetry_mixin.py [hunk 4]
```diff
@@ -70,6 +78,13 @@ def _get_jaeger_exporter(self, parsed_url):
         logger.info('Configuring jaeger exporter..')
         return jaeger_exporter
 
+    def _get_gcp_exporter(self, parsed_url):
```
Comment (3asm) [inline src/ostorlab/agent/mixins/agent_open_telemetry_mixin.py:81]: This requires some documentation one what is expected here, what is the format of the URL, etc.

File: src/ostorlab/agent/mixins/agent_open_telemetry_mixin.py [hunk 5]
```diff
@@ -70,6 +78,13 @@ def _get_jaeger_exporter(self, parsed_url):
...
+        parsed_path = parsed_url.path.split(':')
+        project_id = parsed_path[0]
+        service_account_key = parsed_path[1]
+        os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = service_account_key
```
Comment (3asm) [inline src/ostorlab/agent/mixins/agent_open_telemetry_mixin.py:85]: This warrant a comment on why.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: src/ostorlab/agent/mixins/agent_open_telemetry_mixin.py [hunk 6]
```diff
@@ -42,13 +45,18 @@ def get_trace_exporter(self) -> sdk_export.SpanExporter:
         The urls are customized to respect the following format:
             name_of_the_tracing_tool:hostname:port
             eg: jaeger:jaeger-host:8631
+            for gcp the format is:
+            name_of_the_tracing_tool:project_id:service_account_json_path
+            eg: gcp:project_1:/tmp/key.json
```
Comment (3asm) [inline src/ostorlab/agent/mixins/agent_open_telemetry_mixin.py:50]: we should have the json content in the URL, we should not rely on external paths for passing credentials.

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**